hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/HasSystemContract.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2023-2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.contract.impl.exec.systemcontracts;
+
+import static com.hedera.node.app.service.contract.impl.utils.ConversionUtils.asNumberedContractId;
+
+import com.hedera.hapi.node.base.ContractID;
+import com.hedera.node.app.service.contract.impl.exec.systemcontracts.common.AbstractNativeSystemContract;
+import com.hedera.node.app.service.contract.impl.exec.systemcontracts.has.HasCallFactory;
+import com.hedera.node.app.service.contract.impl.exec.utils.FrameUtils;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.evm.frame.MessageFrame;
+import org.hyperledger.besu.evm.gascalculator.GasCalculator;
+
+@Singleton
+public class HasSystemContract extends AbstractNativeSystemContract implements HederaSystemContract {
+    public static final String HAS_SYSTEM_CONTRACT_NAME = "HAS";
+    public static final String HAS_EVM_ADDRESS = "0x16a";
+    public static final ContractID HAS_CONTRACT_ID = asNumberedContractId(Address.fromHexString(HAS_EVM_ADDRESS));
+
+    @Inject
+    public HasSystemContract(@NonNull final GasCalculator gasCalculator, @NonNull final HasCallFactory callFactory) {
+        super(HAS_SYSTEM_CONTRACT_NAME, callFactory, HAS_CONTRACT_ID, gasCalculator);
+    }
+
+    // Call type is not relevant for the HAS system contracts
```

Comment:
Why is this comment true?  And ... is it true forever or just for our current HAS (+ 632)?

(I guess it's because for HAS it is always the case that sender == authorizer?)

Reply:
Alternatively if it is not used then maybe set it to something invalid, like `null`, in the hopes that if it is ever inadvertently used we find out.